### ISSUE TITLE
Max folder depth is handled incorrectly when importing games

### ISSUE THREAD

**Issue by martyn-harris** [maloep/romcollectionbrowser#240]
There is a bug in the walkDown method of dbupdate.py.  This results in the import Max Folder Depth setting not being correctly applied.

It is probably easier to explain the problem in the code rather than the behaviour of the bug.

While looping through the folders, the maxFolderDepth parameter is decremented for each iteration of the loop.  This is incorrect - it should only be decremented when recursively going into a sub folder.

Here is the code at fault:

```
def walkDown(self, files, romPath, maxFolderDepth):
        Logutil.log("Running walkdown on: %s" %romPath, util.LOG_LEVEL_INFO)

        dirs, newFiles, dirname, filemask = self.getFilesByWildcardExt(romPath)
        files.extend(newFiles)

        for dir in dirs:
            newRomPath = util.joinPath(dirname, dir, filemask)
            maxFolderDepth = maxFolderDepth -1
            if(maxFolderDepth > 0):
                self.walkDown(files, newRomPath, maxFolderDepth)

        return files
```

And my proposed fix:

```
def walkDown(self, files, romPath, maxFolderDepth):
        Logutil.log("Running walkdown on: %s" %romPath, util.LOG_LEVEL_INFO)

        dirs, newFiles, dirname, filemask = self.getFilesByWildcardExt(romPath)
        files.extend(newFiles)

        for dir in dirs:
            newRomPath = util.joinPath(dirname, dir, filemask)
            if(maxFolderDepth > 0):
                self.walkDown(files, newRomPath, (maxFolderDepth -1))

        return files
```

This bug particularly effects people who have their roms organised in folders by game name, for example..

```
    /Roms
        /PlayStation
            /GameName1
                /GameName1.bin
                /GameName1.cue
            /GameName2
                /GameName2.bin
                /GameName2.cue
```

With this setup and the default Max Folder Depth of 99, it will only ever import the first 99 games.

I actually want to set my Max Folder Depth to 1 (to prevent it importing a sub folder containing only foreign games), but obviously it only ever imports the first game in the PlayStation folder.

I have modified my local copy of the file and subsequent imports have worked as expected.

**Comment by martyn-harris** [comment]
I have corrected this in my fork and submitted a pull request.

**Comment by bruny** [comment]
The PR was merged in ba9cc6c
